container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/shared_helpers/shared_helpers.go

```diff
@@ -670,6 +701,95 @@ func GetMatchingUserServiceObjectsAndKubernetesResourcesByServiceName(
 	return matchesByServiceName, nil
 }
 
+// DumpNamespacePods dump pods from the same namespace
+func DumpNamespacePods(
+	ctx context.Context,
+	kubernetesManager *kubernetes_manager.KubernetesManager,
+	namespace *apiv1.Namespace,
+	podsToDump []apiv1.Pod,
+	outputDirpath string,
+) error {
+	// Create output directory
+	if _, err := os.Stat(outputDirpath); !os.IsNotExist(err) {
+		return stacktrace.NewError("Cannot create output directory at '%v'; directory already exists", outputDirpath)
+	}
+	if err := os.Mkdir(outputDirpath, createdDirPerms); err != nil {
+		return stacktrace.Propagate(err, "An error occurred creating output directory at '%v'", outputDirpath)
+	}
+
+	workerPool := workerpool.New(numPodsToDumpAtOnce)
+	resultErrsChan := make(chan dumpPodResult, len(podsToDump))
+	for _, pod := range podsToDump {
+		/*
+			!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! WARNING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+			It's VERY important that the actual `func()` job function get created inside a helper function!!
+			This is because variables declared inside for-loops are created BY REFERENCE rather than by-value, which
+				means that if we inline the `func() {....}` creation here then all the job functions would get a REFERENCE to
+				any variables they'd use.
+			This means that by the time the job functions were run in the worker pool (long after the for-loop finished)
+				then all the job functions would be using a reference from the last iteration of the for-loop.
+
+			For more info, see the "Variables declared in for loops are passed by reference" section of:
+				https://www.calhoun.io/gotchas-and-common-mistakes-with-closures-in-go/ for more details
+			!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! WARNING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+		*/
+		jobToSubmit := createDumpPodJob(
+			ctx,
+			kubernetesManager,
+			namespace.Name,
+			pod,
+			outputDirpath,
+			resultErrsChan,
+		)
+		workerPool.Submit(jobToSubmit)
+	}
+	workerPool.StopWait()
+	close(resultErrsChan)
+
+	resultErrorsByPodName := map[string]error{}
+	for podResult := range resultErrsChan {
+		resultErrorsByPodName[podResult.podName] = podResult.err
+	}
+
+	if len(resultErrorsByPodName) > 0 {
+		combinedErr := BuildCombinedError(resultErrorsByPodName, dumpPodErrorTitle)
+		return combinedErr
+	}
+	return nil
+}
+
+// This is a helper function that will take multiple errors, each identified by an ID, and format them together
+// If no errors are returned, this function returns nil
+func BuildCombinedError(errorsById map[string]error, titleStr string) error {
```

Comment:
we should probably take this out of k8s and put it in some place generic later on